tests/specs/neogit/popups/rebase_spec.lua

```diff
@@ -0,0 +1,46 @@
+local async = require("plenary.async")
+async.tests.add_to_env()
+
+local gl = require("neogit.lib.git.log")
+local gb = require("neogit.lib.git.branch")
+local operations = require("neogit.operations")
+local harness = require("tests.util.git_harness")
+local in_prepared_repo = harness.in_prepared_repo
+
+local FuzzyFinderBuffer = require("tests.mocks.fuzzy_finder")
+local input = require("tests.mocks.input")
+
+local function act(normal_cmd)
+  vim.fn.feedkeys(vim.api.nvim_replace_termcodes(normal_cmd, true, true, true))
+  vim.fn.feedkeys("", "x") -- flush typeahead
+end
+
+describe("rebase popup", function()
+  it(
+    "rebase to reword HEAD~1",
+    in_prepared_repo(function()
+      local new_commit_message = "foobar"
+      local original_branch = gb.current()
+      input.values = { new_commit_message }
+      -- TODO: need to mock `CommitSelectViewBuffer`
```

Comment:
@CKolkey I want to add unit tests for this, but I'm failing to mock the `CommitSelectViewBuffer` =S Do you have any idea how this could be achieved, so that this test can pass the commit to reword into the SUT?

Reply:
...or maybe there is a way to mock `popup.state.env.commit = "HEAD~1"` ?

Reply:
Sure thing - does something like this work?
```lua
-- tests/mocks/commit_select_buffer.lua

local CommitSelectViewBuffer = require("neogit.buffers.commit_select_view")

local M = {
  values = {},
}

function M.add(value)
  table.insert(M.values, value)
end

CommitSelectViewBuffer.open_async = function()
  return table.remove(M.value, 1)
end

return M
```

```lua
-- tests/spec/whatever_spec.lua
local CommitSelectViewBufferMock = require("tests.mocks.commit_select_buffer")

-- ...
it("tests something", in_prepared_repo(function()
    CommitSelectViewBufferMock.add("commit to return")
    -- ...
end))
```

Reply:
> ...or maybe there is a way to mock `popup.state.env.commit = "HEAD~1"` ?

If you wanna do something like that you can do a unit test for the action and just pass a popup table to the function - skip the integration test and treat the action like a function. I _think_ that'll work, unless some coroutine stuff doesn't like that

Reply:
Cool, the `CommitSelectViewBufferMock` worked quite nicely!